Makefile

```diff
@@ -653,10 +654,6 @@ ${PROOFTHEORIES}: %_proofs.thy: %.pbf ${BF_GEN_PATH} ${STATICSOURCES} \
 # Preprocessed source files
 ###########################
 
-%.s: %.S ${GENHEADERS} ${STATICHEADERS} | ${DIRECTORIES}
-	@echo " [CPP] $@"
-	$(Q)${CPP} ${CPPFLAGS} -CC -E -o $@ $<
-
```

Comment:
Yes, given we currently invoke the assembler through gcc the pre-processor is getting invoked anyway. There is a case then to be made for whether we still want to keep the explicit pre-process step or not, but that should be an explicit discussion, not just happen as an unwritten side effect of supporting case insensitive builds.

If you just changed the '.S' to something (like say '.s_pp' if you were to follow naming conventions elsewhere in the Makefile) then we would happily accept that change.

Not saying we wouldn't accept the change of removing a redundant pre-processing step, but it would be an actual discussion.